domain/environment.rb

```diff
@@ -0,0 +1,42 @@
+require 'fakeredis'
+
+class Environment
+  class << self
+    def run(command)
+      @script = command
+    end
+    attr_reader :script
+
+    def redis
+      @redis ||= Redis.new
+    end
+  end
+
+  def script
+    self.class.script
+  end
+
+  def check
+    output = run_script(script)
+    self.class.redis.set("#{self.class.to_s.downcase}_status", exit_status)
```

Comment:
I've checked and ``Redis::Namespace`` plays along with ``FakeRedis`` ok.

I originally thought I needed to maintain just a single ``Redis`` instance for the sake of ``FakeRedis`` and the tests and so I stored it as a class variable on ``Environment`` but this doesn't seem to be necessary so creating a namespaced connection for each subclass should work fine.